exportModule.js

```diff
@@ -12,13 +15,18 @@ function exportModule(data) {
       Terminal: new states.TerminalState('Terminal').toJSON(),
     },
   };
-  dataTypes.forEach((temp, i) => {
-    const stateName = `${temp.substring(21)}_${i}`;
-    const StateClass = factory(temp);
-    if (StateClass !== null) {
-      moduleJSON.states[stateName] = factory(temp).toJSON();
+  dataTypes.forEach((object, i) => {
+    if (object.dataType !== null && object.codes !== undefined && object.type === 'Retrieve') {
+      const link = valueSets[object.codes.name];
+      const StateClass = factory(object.dataType, link);
+      const stateName = `${object.dataType.substring(21)}_${i}`;
+      logger.info(`adding state of type: ${StateClass.name}`);
+      logger.info(`adding value_set with id of: ${link}`);
+      moduleJSON.states[stateName] = StateClass.toJSON();
```

Comment:
I think the flow of this should be as follows:

* If object.dataType is not null, then ok cool we're gonna create a state class
  * Then we check for if a valueset is defined (using that retrieve/object.codes conditional thing that you have here). Even if there is no valueset, we still want to construct a state class. We can either pass in nothing or the link if it exists.